packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmObjectInternal.kt

```diff
@@ -43,130 +27,6 @@ import kotlin.reflect.KClass
  */
 // TODO Public due to being a transative dependency of Mediator
```

Comment:
That dependency goes down to the current sync split. They are needed to be public so they are accessible from the sync module.

We could instead of having base a dependency on Sync, to link the base sources in the sync project. It then would have access to all the internals.